components/datadog/agent/params.go

```diff
@@ -40,18 +42,29 @@ func newParams(env *config.CommonEnvironment, options ...func(*Params) error) (*
 	if env.AgentVersion() != "" {
 		defaultVersion = WithVersion(env.AgentVersion())
 	}
-	if env.PipelineID() != "" {
-		defaultVersion = WithPipelineID(env.PipelineID())
+	versionOptions := []func(*Params) error{defaultVersion}
+
+	// If repository and/or channel are specified, force-set them
+	if env.AgentRepository() != "" {
+		versionOptions = append(versionOptions, WithRepository(env.AgentRepository()))
```

Comment:
For `env.AgentRepository()` I agree. For `env.AgentChannel()`, any string value is valid (I define some `os.Channel` constants for readability and to avoid repetition, but any value can be a valid channel, and I don't want to restrict users to only the channels that are explicitly defined in code).